cms/djangoapps/api/docs/decisions/001-course-apps-api.rst

```diff
@@ -0,0 +1,153 @@
+platform and are inseparable from it, however future apps could be implemented
+as plugins.
+
+In order to drive this new UI, we need a way to discover installed and available
+course apps and provide an API that can be used by MFEs to display information
+related to these apps and potentially enable/disable these apps.
+
+.. _Course Authoring MFE: https://github.com/edx/frontend-app-course-authoring/
+
+
+Decision
+========
+
+We can develop a new course apps API that can be used to manage the apps
+available for a course. The list of *available* apps might not match the list
+of *installed* apps since it is possible that certain apps may be disabled, or
+otherwise unavailable for a particular course.
+
+The course apps API will only list the available apps, and allow toggling them
+on or off. It will not support any other verbs or settings. Each app can
+provide its own API(s) for configuring app-specific settings, which will not be
+in scope here. Currently the mechanism for extending the frontend with custom
+config pages for plug-in apps is TBD.
+
+Each app will have the following associated metadata:
+
+- **id**: A unique identifier for the app. This can be used in the frontend to
+  look up the app's friendly title and description.
+- **enabled**: Is this app enabled for the current course.
+- **permissions**: Apps can potentially enable/disable certain operations, for
+  users based on their role. The permissions a user has can be listed here. Apps
+  may have their own permissions but at this very least this will include:
+
+    - **enable**: Can the current user enable/disable this app.
+    - **configure**: Can the current user can configure this app.
+
+  Apps may have their own custom fine-grained controls over what settings a
+  particular user can configure.
+
+  If an app doesn't have any configuration it can leave out the ``configure``
+  key and the UI will simply not show a configuration option for that app.
+- **legacy_link**: If available, this will point to the legacy link for
+  configuring the app. This can be provided as a fallback while the new UX is
+  still in development.
+
+Here is the structure that will form the basis of the API's response:
+
+.. code-block:: python
+
+    {
+        'id': 'courseapp',
+        'enabled': False,
+        'permissions': {
+            'enable': True,
+            'configure': True,
+            'edit_lti_config': False,
+        },
+        'legacy_link': 'https://studio.example.com/course_id/app-page',
+    }
+
+
+This API can be hosted at: ``/course_apps/v1/apps/{course_id}/``
+
+A ``GET`` request to this API will return an array of objects with the above
+structure. A ``PATCH`` request to the same endpoint with just the ``id`` and the
+`enabled` attribute can be used to enable/disable the app if the user has the
+requisite permissions.
+
+Each app will also have a user-friendly title and description, however this will
```

Comment:
After coding up this same sort of thing in the frontend for our discussions apps, I'm not sure it's what we actually want.  It means that frontend-app-course-authoring needs to have a hard-coded list of app i18n messages, which feels a bit odd.  Ideally the frontend would be agnostic to the exact apps its displaying, I'd expect.  

Long term, I think it makes sense for the source of truth for the apps to also be the source of truth for their localizations.

Reply:
Sure. I can add back the title and description here.